app/change_set_persisters/plum_change_set_persister/apply_visibility_by_date.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+class PlumChangeSetPersister
+  class ApplyVisibilityByDate
+    attr_reader :change_set_persister, :change_set
+    def initialize(change_set_persister:, change_set:, post_save_resource: nil)
+      @change_set = change_set
+      @change_set_persister = change_set_persister
+    end
+
+    def run
+      return unless change_set.try(:set_visibility_by_date?)
+      return unless date_object
+      change_set.validate(visibility: calculated_visibility, rights_statement: calculated_rights_statement)
+      change_set.sync
+      change_set
+    end
+
+    private
+
+      def date
+        Array.wrap(change_set.resource.primary_imported_metadata.created).first
+      end
+
+      def date_object
+        return unless date
+        Time.zone.parse(date)
+      rescue
+        Rails.logger.debug("Unable to parse created date: #{change_set.resource.primary_imported_metadata.created}")
```

Comment:
Is `debug` normally used for these types of errors, or would `warn` be needed? (I'm uncertain as to whether or not the logging level might affect this).

Reply:
Switched it to warn.